### ISSUE TITLE
Enhanced Debug Logging ERROR - Passed type is <class 'list'> and length is 1

### ISSUE THREAD

**Issue by purcell-lab** [davidusb-geek/emhass#350]
**Describe the bug**
Sometimes when getting obscure data errors, it is very difficult to determine which array is in error:

```
2024-10-04 20:21:07,628 - web_server - INFO - Passed runtime parameters: {'load_cost_forecast': [0.36, 0.34, 0.21, 0.2, 0.2, 0.2, 0.19, 0.19, 0.18, 0.18, 0.18, 0.18, 0.18, 0.17, 0.17, 0.17, 0.17, 0.17, 0.18, 0.18, 0.16, 0.12, 0.08, 0.07, 0.07, 0.07, 0.04, 0.04, 0.04, 0.04, 0.04, 0.04, 0.04, 0.04, 0.04, 0.04, 0.04, 0.05, 0.06, 0.08, 0.26, 0.31, 0.34, 0.36, 0.38, 0.37, 0.35, 0.34, 0.34], 'prod_price_forecast': [0.11, 0.09, 0.09, 0.08, 0.08, 0.08, 0.07, 0.07, 0.06, 0.06, 0.06, 0.06, 0.06, 0.06, 0.05, 0.05, 0.05, 0.05, 0.06, 0.06, 0.05, 0.01, -0.03, -0.04, -0.04, -0.04, -0.04, -0.04, -0.04, -0.05, -0.04, -0.04, -0.04, -0.04, -0.04, -0.04, -0.04, -0.04, -0.03, -0.01, 0.02, 0.06, 0.09, 0.11, 0.12, 0.12, 0.1, 0.1, 0.09], 'load_power_forecast': [347, 500, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 700, 700, 1200, 400, 400, 300, 300, 300, 300, 800, 1000, 1000, 2000, 1000, 500, 500, 500, 600, 600, 3100, 3100, 300, 300, 300, 300, 300, 300, 300, 300, 600, 700, 700, 1700, 1200, 800, 600, 400, 500, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 700, 700, 1200, 400, 400, 300, 300, 300, 300, 800, 1000, 1000, 2000, 1000, 500, 500, 500, 600, 600, 3100, 3100, 300, 300, 300, 300, 300, 300, 300, 300, 600, 700, 700, 1700, 1200, 800, 600], 'pv_power_forecast': [0], 'prediction_horizon': 48, 'def_load_config': [{}, {}, {}, {'thermal_config': {'heating_rate': -5.0, 'sense': 'cool', 'cooling_constant': 0.2, 'overshoot_temperature': 16.0, 'start_temperature': 17.3, 'desired_temperatures': [24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24]}}, {}, {}], 'outdoor_temperature_forecast': [17, 17, 16, 16, 15, 15, 15, 15, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 15, 15, 18, 18, 21, 21, 23, 23, 25, 25, 27, 27, 27, 27, 28, 28, 28, 28, 28, 28, 27, 27, 25, 25, 23, 23, 21, 21], 'alpha': 1, 'beta': 0, 'num_def_loads': 6, 'def_total_hours': [0, 9, 0, 8, 9, 4], 'def_end_timestep': [0, 0, 0, 0, 0, 0], 'P_deferrable_nom': [300, 6333, 11520, 4500, 600, 11520], 'def_current_state': [0, 0, 0, 0, 0, 0], 'def_start_penalty': [1, 1, 1, 1, 1, 1], 'treat_def_as_semi_cont': [1, 1, 0, 0, 1, 0], 'set_def_constant': [0, 0, 0, 0, 0, 0], 'soc_init': 0.16, 'soc_final': 1}
2024-10-04 20:21:07,628 - web_server - INFO -  >> Setting input data dict
2024-10-04 20:21:07,628 - web_server - INFO - Setting up needed data
2024-10-04 20:21:07,631 - web_server - ERROR - ERROR: The passed data is either not a list or the length is not correct, length should be 48
2024-10-04 20:21:07,631 - web_server - ERROR - Passed type is <class 'list'> and length is 1
```

I suspect this is related to the @werdnum thermal model, but unsure which list is supposed to be length 48. 

Is it possible to increase the detailed provided in debug logging statements to expose where these types of errors are hidden?

**Comment by davidusb-geek** [comment]
Well in this case it seems to be this isn't it?
```
'pv_power_forecast': [0]
```

**Comment by purcell-lab** [comment]
Thanks, that is the solution, but my request for more debug details is still valid I think..

I'm happy to raise some PR's once I can figure out the dev environment setup.

**Comment by davidusb-geek** [comment]
More logging was recently added, closing this issue

**Comment by purcell-lab** [comment]
Debug Logging included in 0.13 is excellent for weather and pv forecast, thanks.

```logs
[2025-04-03 07:21:21 +1000] [22] [DEBUG] get_weather_forecast returning:
                           temp_air  relative_humidity  ...    dhi   dni
2025-04-03 07:00:00+10:00      23.5                 95  ...   40.1  11.8
2025-04-03 07:30:00+10:00      23.5                 95  ...   50.0  17.9
2025-04-03 08:00:00+10:00      23.6                 95  ...   81.7  28.6
2025-04-03 08:30:00+10:00      24.0                 94  ...  140.1  45.5
2025-04-03 09:00:00+10:00      24.5                 92  ...  191.3  71.5
...                             ...                ...  ...    ...   ...
2025-04-05 04:30:00+10:00      23.5                 89  ...    0.0   0.0
2025-04-05 05:00:00+10:00      23.4                 89  ...    0.0   0.0
2025-04-05 05:30:00+10:00      23.4                 89  ...    0.0   0.0
2025-04-05 06:00:00+10:00      23.4                 89  ...    0.0   0.0
2025-04-05 06:30:00+10:00      23.4                 89  ...   17.9  66.5
[96 rows x 8 columns]
[2025-04-03 07:21:25 +1000] [22] [DEBUG] get_power_from_weather returning:
2025-04-03 07:00:00+10:00     634.000000
2025-04-03 07:30:00+10:00     869.113301
2025-04-03 08:00:00+10:00    1552.478316
2025-04-03 08:30:00+10:00    2783.202684
2025-04-03 09:00:00+10:00    3986.363589
                                ...     
2025-04-05 04:30:00+10:00       0.000000
2025-04-05 05:00:00+10:00       0.000000
2025-04-05 05:30:00+10:00       0.000000
2025-04-05 06:00:00+10:00       0.000000
2025-04-05 06:30:00+10:00     457.253033
Freq: 30min, Length: 96, dtype: float64
```
